org.eclipse.jdt.core/model/org/eclipse/jdt/core/compiler/CompilerConfiguration.java

```diff
@@ -0,0 +1,169 @@
+/*******************************************************************************
+* Copyright (c) 2024 Microsoft Corporation and others.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License 2.0
+* which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-2.0/
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Microsoft Corporation - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.jdt.core.compiler;
+
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.jdt.internal.compiler.impl.CompilerOptions;
+
+/**
+ * This class encapsulates the standard compiler options that can be
+ * used to compile Java files. It provides methods to set and retrieve
+ * various compiler options, including source paths, class paths,
+ * output directories, annotation processing options, and other compiler
+ * arguments.
+ *
+ * Clients typically use this class when opting for an alternative compiler
+ * like Javac to compile Java files.
+ *
+ * @since 3.38
+ * @noextend This class is not intended to be subclassed by clients.
+ */
+public class CompilerConfiguration {
+	List<String> sourcepaths;
+	List<String> moduleSourcepaths;
+	List<String> classpaths;
+	List<String> modulepaths;
+	Map<File, File> sourceOutputMapping;
```

Comment:
I am worried, that this otherwise unneeded API 

- became that big,
- is not tested at all, 
- the implementing members are not private, 
- it is not threadsafe. 
- publishes internal Classes

Feels like it should be an immutable record.

The IDE even warns about it:
![image](https://github.com/eclipse-jdt/eclipse.jdt.core/assets/51790620/c958ee25-4d0b-42c5-931b-67b6a43c23af)

Reply:
Yes, it's cleaner to use record for this API. Also add unit test to cover the new API.